lib/screens/conversations_screen.dart

```diff
@@ -1,20 +1,61 @@
 import 'package:chirp_nets/providers/users.dart';
+import 'package:chirp_nets/screens/compass_screen.dart';
 import 'package:chirp_nets/screens/settings_screen.dart';
 import 'package:chirp_nets/widgets/conversations/add_conversation_widget.dart';
 import 'package:chirp_nets/widgets/conversations/conversations_list_widget.dart';
 import 'package:chirp_nets/widgets/users/add_first_user_widget.dart';
 import 'package:flutter/material.dart';
+import 'package:flutter/services.dart';
+import 'package:location/location.dart';
 import 'package:provider/provider.dart';
 
 import 'package:chirp_nets/models/conversation.dart';
 import 'package:chirp_nets/models/user.dart';
 import 'package:chirp_nets/providers/conversations.dart';
 
-class ConversationsScreen extends StatelessWidget {
-  // Wait until page is loaded and check if a user exits
-  // If no user exists prompt the user to create
+class ConversationsScreen extends StatefulWidget {
+  ConversationsScreen();
+
+  @override
+  _ConversationsScreenState createState() => _ConversationsScreenState();
+}
+
+class _ConversationsScreenState extends State<ConversationsScreen> {
+  Users userData;
+
+  var location = new Location();
+
+  fetchLocation(LocationData currentLocation) async {
+    try {
+      currentLocation = await location.getLocation();
+    } on PlatformException catch (e) {
+      if (e.code == 'PERMISSION_DENIED') {
+        var error = 'Permission denied';
+        debugPrint(error);
+      }
+      currentLocation = null;
+    }
+    if (currentLocation != null && userData != null) {
+      if (userData.currentUser != null) {
+        userData.updateLocation(
+          id: userData.currentUser.id,
+          latitude: currentLocation.latitude,
+          longitude: currentLocation.longitude,
+        );
+      }
+    }
+  }
+
+  @override
+  void initState() {
+    super.initState();
+    location.onLocationChanged().listen((LocationData currentLocation) {
+      fetchLocation(currentLocation);
+    });
+  }
+
   void getBottomSheet(userData, context, conversationData) {
-    User user = userData.getCurrentUser();
+    User user = userData.currentUser;
```

Comment:
Why was this changed from a getter?

Reply:
It was changed from a function to a local private variable, we need the user in a lot of places so it makes more sense to store them than to fetch them every time. The added line is the getter syntax for class variables

Reply:
It seems like you are directly accessing the variable, thus exposing it to caller. So any caller could alter and do whatever they like to the CurrentUser field. Using getters and setters for access and changes allows us to add safety checks and control the access to this variable. 

Am I misunderstanding?

Reply:
Since its private it needs a setter to update the variable. The only way to do that is to call the updateUser method which doesn't allow updating the currentUser field or to update the database manually. All changes to models are handled by the providers for the reasons you gave above

Reply:
Ah okay, I understand now. I did not know that flutter handled getters with the "get" keyboard making them look like variables.